diff -ur CHOLMOD/Demo/Makefile CHOLMOD-uf/Demo/Makefile
--- CHOLMOD/Demo/Makefile	2007-05-05 16:37:46.000000000 +0200
+++ CHOLMOD-uf/Demo/Makefile	2007-12-02 17:10:39.000000000 +0100
@@ -7,7 +7,7 @@
 
 default: all
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #-------------------------------------------------------------------------------
 # With METIS, CCOLAMD, CAMD, and the Partition Module:
@@ -64,7 +64,7 @@
 # 	( cd ../../COLAMD ; $(MAKE) )
 #-------------------------------------------------------------------------------
 
-I = -I../Include -I../../SuiteSparse_config
+I = -I../Include
 
 cholmod_demo: library cholmod_demo.c cholmod_demo.h
 	$(C) -o cholmod_demo $(I) cholmod_demo.c $(LIB2)
diff -ur CHOLMOD/Doc/Makefile CHOLMOD-uf/Doc/Makefile
--- CHOLMOD/Doc/Makefile	2007-05-03 23:32:54.000000000 +0200
+++ CHOLMOD-uf/Doc/Makefile	2007-12-02 17:10:52.000000000 +0100
@@ -1,6 +1,6 @@
 default: all
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 all: UserGuide.pdf
 
--- CHOLMOD/Include/cholmod.h.orig	2012-05-29 18:45:58.000000000 +0200
+++ CHOLMOD/Include/cholmod.h	2012-06-27 18:02:23.815293493 +0200
@@ -83,7 +83,7 @@
 /* assume large file support.  If problems occur, compile with -DNLARGEFILE */
 #include "cholmod_io64.h"
 
-#include "SuiteSparse_config.h"
+#include <SuiteSparse_config.h>
 
 #include "cholmod_config.h"
 
diff -ur CHOLMOD/Include/cholmod_internal.h CHOLMOD-uf/Include/cholmod_internal.h
--- CHOLMOD/Include/cholmod_internal.h	2007-09-28 20:31:04.000000000 +0200
+++ CHOLMOD-uf/Include/cholmod_internal.h	2007-12-02 17:13:05.000000000 +0100
@@ -201,7 +201,7 @@
  * It can also be redefined for other platforms, by modifying SuiteSparse_config.h.
  */
 
-#include "SuiteSparse_config.h"
+#include <SuiteSparse_config.h>
 
 /* -------------------------------------------------------------------------- */
 /* Size_max: the largest value of size_t */
--- CHOLMOD/Lib/Makefile.orig	2011-12-14 18:17:26.000000000 +0100
+++ CHOLMOD/Lib/Makefile	2012-01-01 17:07:06.246620933 +0100
@@ -6,7 +6,7 @@
 
 ccode: all
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #-------------------------------------------------------------------------------
 # the optional Partition module requires METIS, CAMD, and CCOLAMD
@@ -14,9 +14,9 @@
 CONFIG =
 ifeq (,$(findstring -DNPARTITION, $(CHOLMOD_CONFIG)))
     # METIS is check if METIS is available
-    ifeq (../../metis-4.0, $(wildcard ../../metis-4.0))
-        I_WITH_PARTITION = -I$(METIS_PATH)/Lib \
-            -I../../CCOLAMD/Include -I../../CAMD/Include
+    ifeq ($(WITH_METIS), 1)
+        I_WITH_PARTITION = -I/usr/include/metis \
+            -I/usr/include/ccolamd -I/usr/include/camd
     else
         # METIS is not available, but CHOLMOD_CONFIG does not specify
         # -DNPARTITION.  Add it here so we can still compile CHOLMOD.
@@ -24,8 +24,8 @@
     endif
 endif
 
-I = -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include \
-	$(I_WITH_PARTITION) -I../Include -I../../SuiteSparse_config
+I = -I/usr/include/amd -I/usr/include/colamd \
+	$(I_WITH_PARTITION) -I../Include
 
 #-------------------------------------------------------------------------------
 
diff -ur CHOLMOD/Makefile CHOLMOD-uf/Makefile
--- CHOLMOD/Makefile	2006-11-30 17:40:08.000000000 +0100
+++ CHOLMOD-uf/Makefile	2007-12-02 17:12:17.000000000 +0100
@@ -7,7 +7,7 @@
 
 default: all
 
-include ../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 # Compile the C-callable libraries and the Demo programs.
 all:
diff -ur CHOLMOD/MATLAB/cholmod_matlab.h CHOLMOD-uf/MATLAB/cholmod_matlab.h
--- CHOLMOD/MATLAB/cholmod_matlab.h	2007-06-14 19:52:47.000000000 +0200
+++ CHOLMOD-uf/MATLAB/cholmod_matlab.h	2007-12-02 17:13:28.000000000 +0100
@@ -4,7 +4,7 @@
 
 /* Shared prototypes and definitions for CHOLMOD mexFunctions */
 
-#include "SuiteSparse_config.h"
+#include <SuiteSparse_config.h>
 #ifndef DLONG
 #define DLONG
 #endif
diff -ur CHOLMOD/Tcov/Makefile CHOLMOD-uf/Tcov/Makefile
--- CHOLMOD/Tcov/Makefile	2007-05-05 19:12:46.000000000 +0200
+++ CHOLMOD-uf/Tcov/Makefile	2007-12-02 17:11:45.000000000 +0100
@@ -5,7 +5,7 @@
 # If you compile CHOLMOD with -DNPARTITION, then you do not need METIS,
 # CCOLAMD, or the Partition module.
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 include Make.inc
 
 # Tcov requires gcc
@@ -31,7 +31,7 @@
 
 I = -I../../AMD/Include -I../../COLAMD/Include \
 	-I$(METIS_PATH)/Lib -I../../CCOLAMD/Include -I../../CAMD/Include \
-	-I../Include -I../../SuiteSparse_config
+	-I../Include
 
 default: cm cl z_demo l_demo cmread clread
 
