diff -ur CHOLMOD/Demo/Makefile CHOLMOD.shared/Demo/Makefile
--- CHOLMOD/Demo/Makefile	2007-12-07 12:29:51.000000000 +0100
+++ CHOLMOD.shared/Demo/Makefile	2007-12-07 12:20:59.000000000 +0100
@@ -11,12 +11,8 @@
 
 #-------------------------------------------------------------------------------
 # With METIS, CCOLAMD, CAMD, and the Partition Module:
-LIB2 = ../Lib/libcholmod.a ../../AMD/Lib/libamd.a ../../COLAMD/Lib/libcolamd.a \
-	../../CCOLAMD/Lib/libccolamd.a ../../CAMD/Lib/libcamd.a \
-	$(METIS) $(LAPACK) $(BLAS) $(XERBLA) $(LIB)
-# Use this instead, if you compile with -DNPARTITION:
-# LIB2 = ../Lib/libcholmod.a ../../AMD/Lib/libamd.a ../../COLAMD/libcolamd.a \
-	$(LAPACK) $(BLAS) $(XERBLA) $(LIB)
+LIB2 = ../Lib/libcholmod.la -lamd -lcolamd -lccolamd -lcamd \
+	$(METIS) $(LAPACK) $(BLAS) $(LIB)
 #-------------------------------------------------------------------------------
 
 C = $(CC) $(CFLAGS) $(CHOLMOD_CONFIG)
@@ -48,32 +44,20 @@
 	- $(RM) $(CLEAN)
 
 #-------------------------------------------------------------------------------
-# See below if you compile with -DNPARTITION
 library:
-	( cd ../../UFconfig/xerbla ; $(MAKE) )
 	( cd ../Lib ; $(MAKE) )
-	( cd ../../AMD ; $(MAKE) library )
-	( cd ../../CAMD ; $(MAKE) library )
-	( cd ../../COLAMD ; $(MAKE) library )
-	( cd ../../CCOLAMD ; $(MAKE) library )
-# use this rule instead, if you compile with -DNPARTITION:
-# library:
-# 	( cd ../../UFconfig/xerbla ; $(MAKE) )
-# 	( cd ../Lib ; $(MAKE) )
-# 	( cd ../../AMD ; $(MAKE) library )
-# 	( cd ../../COLAMD ; $(MAKE) )
 #-------------------------------------------------------------------------------
 
 I = -I../Include
 
 cholmod_demo: library cholmod_demo.c cholmod_demo.h
-	$(C) -o cholmod_demo $(I) cholmod_demo.c $(LIB2)
+	libtool --tag=CC --mode=link $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) -I../Include -o cholmod_demo cholmod_demo.c $(LIB2)
 
 cholmod_simple: library cholmod_simple.c
-	$(C) -o cholmod_simple $(I) cholmod_simple.c $(LIB2)
+	libtool --tag=CC --mode=link $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) -I../Include -o cholmod_simple cholmod_simple.c $(LIB2)
 
 cholmod_l_demo: library cholmod_l_demo.c cholmod_demo.h
-	$(C) -o cholmod_l_demo $(I) cholmod_l_demo.c $(LIB2)
+	libtool --tag=CC --mode=link $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) -I../Include -o cholmod_l_demo cholmod_l_demo.c $(LIB2)
 
 readhb: readhb.f
 	$(F77) $(FFLAGS) -o readhb readhb.f
diff -ur CHOLMOD/Lib/Makefile CHOLMOD.shared/Lib/Makefile
--- CHOLMOD/Lib/Makefile	2007-12-07 12:29:51.000000000 +0100
+++ CHOLMOD.shared/Lib/Makefile	2007-12-07 12:29:12.000000000 +0100
@@ -10,14 +10,14 @@
 
 C = $(CC) $(CFLAGS) $(CHOLMOD_CONFIG)
 
-all: libcholmod.a
+all: libcholmod.la
 
-library: libcholmod.a
+library: libcholmod.la
 
 purge: distclean
 
 distclean: clean
-	- $(RM) libcholmod.a
+	- $(RM) libcholmod.la
 
 clean:
 	- $(RM) $(CLEAN)
@@ -109,39 +109,41 @@
 # to compile just the double/int version, use OBJ = $(DI)
 OBJ = $(DI) $(DL)
 
-libcholmod.a: $(OBJ)
-	$(AR) libcholmod.a $(OBJ)
-	$(RANLIB) libcholmod.a
+libcholmod.la: $(OBJ)
+	libtool --tag=CC --mode=link $(CC) $(LDFLAGS) -o libcholmod.la -rpath $(libdir) $(OBJ) -lm \
+		-lamd -lcolamd -lccolamd -lcamd $(METIS) $(LAPACK) $(BLAS)
+
+install: libcholmod.la
+	install -d $(DESTDIR)$(libdir)
+	libtool --mode=install cp libcholmod.la $(DESTDIR)$(libdir)/libcholmod.la
 
 $(OBJ): $(INC)
 
-I = -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include \
-	-I$(METIS_PATH)/Lib -I../../CCOLAMD/Include -I../../CAMD/Include \
-	-I../Include
+I = -I../Include -I$(METIS_PATH)
 
 #-------------------------------------------------------------------------------
 # Check Module:
 #-------------------------------------------------------------------------------
 
 cholmod_check.o: ../Check/cholmod_check.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_read.o: ../Check/cholmod_read.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_write.o: ../Check/cholmod_write.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 #-------------------------------------------------------------------------------
 
 cholmod_l_check.o: ../Check/cholmod_check.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_read.o: ../Check/cholmod_read.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_write.o: ../Check/cholmod_write.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 
 #-------------------------------------------------------------------------------
@@ -149,92 +150,92 @@
 #-------------------------------------------------------------------------------
 
 cholmod_common.o: ../Core/cholmod_common.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_dense.o: ../Core/cholmod_dense.c ../Core/t_cholmod_dense.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_factor.o: ../Core/cholmod_factor.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_change_factor.o: ../Core/cholmod_change_factor.c \
 	../Core/t_cholmod_change_factor.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_memory.o: ../Core/cholmod_memory.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_sparse.o: ../Core/cholmod_sparse.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_complex.o: ../Core/cholmod_complex.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_transpose.o: ../Core/cholmod_transpose.c ../Core/t_cholmod_transpose.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_band.o: ../Core/cholmod_band.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_copy.o: ../Core/cholmod_copy.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_triplet.o: ../Core/cholmod_triplet.c ../Core/t_cholmod_triplet.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_error.o: ../Core/cholmod_error.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_aat.o: ../Core/cholmod_aat.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_add.o: ../Core/cholmod_add.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 #-------------------------------------------------------------------------------
 
 cholmod_l_common.o: ../Core/cholmod_common.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_dense.o: ../Core/cholmod_dense.c ../Core/t_cholmod_dense.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_factor.o: ../Core/cholmod_factor.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_change_factor.o: ../Core/cholmod_change_factor.c \
 	../Core/t_cholmod_change_factor.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_memory.o: ../Core/cholmod_memory.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_sparse.o: ../Core/cholmod_sparse.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_complex.o: ../Core/cholmod_complex.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_transpose.o: ../Core/cholmod_transpose.c ../Core/t_cholmod_transpose.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_band.o: ../Core/cholmod_band.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_copy.o: ../Core/cholmod_copy.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_triplet.o: ../Core/cholmod_triplet.c ../Core/t_cholmod_triplet.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_error.o: ../Core/cholmod_error.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_aat.o: ../Core/cholmod_aat.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_add.o: ../Core/cholmod_add.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 
 #-------------------------------------------------------------------------------
@@ -242,116 +243,116 @@
 #-------------------------------------------------------------------------------
 
 cholmod_amd.o: ../Cholesky/cholmod_amd.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_analyze.o: ../Cholesky/cholmod_analyze.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_colamd.o: ../Cholesky/cholmod_colamd.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_etree.o: ../Cholesky/cholmod_etree.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_factorize.o: ../Cholesky/cholmod_factorize.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_postorder.o: ../Cholesky/cholmod_postorder.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_rcond.o: ../Cholesky/cholmod_rcond.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_resymbol.o: ../Cholesky/cholmod_resymbol.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_rowcolcounts.o: ../Cholesky/cholmod_rowcolcounts.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_solve.o: ../Cholesky/cholmod_solve.c ../Cholesky/t_cholmod_lsolve.c \
 	../Cholesky/t_cholmod_ltsolve.c ../Cholesky/t_cholmod_solve.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_spsolve.o: ../Cholesky/cholmod_spsolve.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_rowfac.o: ../Cholesky/cholmod_rowfac.c ../Cholesky/t_cholmod_rowfac.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 #-------------------------------------------------------------------------------
 
 cholmod_l_amd.o: ../Cholesky/cholmod_amd.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_analyze.o: ../Cholesky/cholmod_analyze.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_colamd.o: ../Cholesky/cholmod_colamd.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_etree.o: ../Cholesky/cholmod_etree.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_factorize.o: ../Cholesky/cholmod_factorize.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_postorder.o: ../Cholesky/cholmod_postorder.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_rcond.o: ../Cholesky/cholmod_rcond.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_resymbol.o: ../Cholesky/cholmod_resymbol.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_rowcolcounts.o: ../Cholesky/cholmod_rowcolcounts.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_solve.o: ../Cholesky/cholmod_solve.c ../Cholesky/t_cholmod_lsolve.c \
 	../Cholesky/t_cholmod_ltsolve.c ../Cholesky/t_cholmod_solve.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_spsolve.o: ../Cholesky/cholmod_spsolve.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_rowfac.o: ../Cholesky/cholmod_rowfac.c ../Cholesky/t_cholmod_rowfac.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # Partition Module:
 #-------------------------------------------------------------------------------
 
 cholmod_ccolamd.o: ../Partition/cholmod_ccolamd.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_csymamd.o: ../Partition/cholmod_csymamd.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_camd.o: ../Partition/cholmod_camd.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_metis.o: ../Partition/cholmod_metis.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_nesdis.o: ../Partition/cholmod_nesdis.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 #-------------------------------------------------------------------------------
 
 cholmod_l_ccolamd.o: ../Partition/cholmod_ccolamd.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_csymamd.o: ../Partition/cholmod_csymamd.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_camd.o: ../Partition/cholmod_camd.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_metis.o: ../Partition/cholmod_metis.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_nesdis.o: ../Partition/cholmod_nesdis.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 
 #-------------------------------------------------------------------------------
@@ -359,88 +360,88 @@
 #-------------------------------------------------------------------------------
 
 cholmod_horzcat.o: ../MatrixOps/cholmod_horzcat.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_norm.o: ../MatrixOps/cholmod_norm.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_scale.o: ../MatrixOps/cholmod_scale.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_drop.o: ../MatrixOps/cholmod_drop.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_sdmult.o: ../MatrixOps/cholmod_sdmult.c \
 	../MatrixOps/t_cholmod_sdmult.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_ssmult.o: ../MatrixOps/cholmod_ssmult.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_submatrix.o: ../MatrixOps/cholmod_submatrix.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_vertcat.o: ../MatrixOps/cholmod_vertcat.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_symmetry.o: ../MatrixOps/cholmod_symmetry.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 #-------------------------------------------------------------------------------
 
 cholmod_l_horzcat.o: ../MatrixOps/cholmod_horzcat.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_norm.o: ../MatrixOps/cholmod_norm.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_scale.o: ../MatrixOps/cholmod_scale.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_drop.o: ../MatrixOps/cholmod_drop.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_sdmult.o: ../MatrixOps/cholmod_sdmult.c \
 	../MatrixOps/t_cholmod_sdmult.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_ssmult.o: ../MatrixOps/cholmod_ssmult.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_submatrix.o: ../MatrixOps/cholmod_submatrix.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_vertcat.o: ../MatrixOps/cholmod_vertcat.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_symmetry.o: ../MatrixOps/cholmod_symmetry.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # Modify Module:
 #-------------------------------------------------------------------------------
 
 cholmod_rowadd.o: ../Modify/cholmod_rowadd.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_rowdel.o: ../Modify/cholmod_rowdel.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_updown.o: ../Modify/cholmod_updown.c \
 	../Modify/t_cholmod_updown.c ../Modify/t_cholmod_updown_numkr.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 #-------------------------------------------------------------------------------
 
 cholmod_l_rowadd.o: ../Modify/cholmod_rowadd.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_rowdel.o: ../Modify/cholmod_rowdel.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_updown.o: ../Modify/cholmod_updown.c \
 	../Modify/t_cholmod_updown.c ../Modify/t_cholmod_updown_numkr.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 
 #-------------------------------------------------------------------------------
@@ -449,24 +450,24 @@
 
 cholmod_super_numeric.o: ../Supernodal/cholmod_super_numeric.c \
 	../Supernodal/t_cholmod_super_numeric.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_super_symbolic.o: ../Supernodal/cholmod_super_symbolic.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 cholmod_super_solve.o: ../Supernodal/cholmod_super_solve.c \
 	../Supernodal/t_cholmod_super_solve.c
-	$(C) -c $(I) $<
+	libtool --tag=CC --mode=compile $(CC) $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $<
 
 #-------------------------------------------------------------------------------
 
 cholmod_l_super_numeric.o: ../Supernodal/cholmod_super_numeric.c \
 	../Supernodal/t_cholmod_super_numeric.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_super_symbolic.o: ../Supernodal/cholmod_super_symbolic.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
 
 cholmod_l_super_solve.o: ../Supernodal/cholmod_super_solve.c \
 	../Supernodal/t_cholmod_super_solve.c
-	$(C) -DDLONG -c $(I) $< -o $@
+	libtool --tag=CC --mode=compile $(CC) -DDLONG $(CFLAGS) $(CHOLMOD_CONFIG) $(I) -c $< -o $@
